barretenberg/cpp/src/barretenberg/honk/proof_system/logderivative_library.hpp

```diff
@@ -31,6 +31,18 @@ void compute_logderivative_inverse(Polynomials& polynomials, auto& relation_para
     constexpr size_t WRITE_TERMS = Relation::WRITE_TERMS;
 
     auto lookup_relation = Relation();
+
+    // SELF: we get the inverse polynomial from the lookup relation template
+    // - > The numerator in any of these will be 1, therefore we have to make an addition to the denominator for the row
+    // -> Rather what we do is add all of the denominators in the inverse polynomial, then we do a batch inversion so
+    // that the whole polynomial consists of inversions QUESTION: the doc comment seems to imply that the inverse is
```

Comment:
A read_term is a value computed as a function of columns and challenges that represents the entry that we are looking up. For example it can be `\beta * s_box_index + s_box_value + \gamma`.
A write term is a value that represent a lookup entry being added to the lookup table.
The inverse polynomial consists of entries which are either:
1. Zero if no lookup or additions of entries to the lookup table are performed in this row
2. The inverse of products of all read and write terms in this row (even those we are not using in the row) if at least one read or write term is being used in this row.

In a nutshell, even 1 term being used enforces the full computation at row.